docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,583 @@
+
+Wildcard entries cannot be declared `optional` or `envfree`, since these
+annotations only make sense for specific contract methods.
+
+```{warning}
+The meaning of your summarizations will change from CVL 1 to CVL 2.  In CVL 2,
+any entry without a `_` will only apply to a single contract!
+```
+
+### Requirements on `returns`
+
+In CVL 2, methods block entries require a `returns` clause in the following
+situations:
+
+ - The method is not a wildcard entry and the contract function returns a value.
+   In this case, the methods block entry must match the contract function's
+   return type.
+
+   ```{todo}
+   Error message
+   ```
+
+ - The method is a wildcard entry that is summarized with a ghost or CVL
+   function summary.  In this case, the return type from the CVL function must
+   be compatible with the declared return type for the method.
+
+   ```{todo}
+   Error message
+   ```
+
+In all other situations, a `returns` clause is forbidden.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+Changes to integer types
+------------------------
+
+In CVL 1, the rules for casting between integer types were complex; CVL 2
+simplifies them.
+
+The general rule of thumb is that you should use `mathint` for all function
+outputs, and the appropriate `int` or `uint` type for all function inputs.
+
+It is now impossible for CVL math expressions to cause overflow - all integer
+operations are exact.
+
+### Mathematical operations return `mathint`
+
+In CVL 2, the result of all arithmetic operators have type `mathint`,
+regardless of the input types.  Arithmetic operators include `+`,
+`*`, `-`, `/`, `^`, and `%`, but not bitwise operators like `<<`, `xor`, and `|`
+(changes to bitwise operators are described {ref}`below <cvl2-bitwise>`).
+
+The primary impact of this change is that you may need to declare more of your
+variables as `mathint` instead of `uint`.  If you are performing arithmetic
+operations and integers that you are passing to specs, you will need to be more
+explicit about the overflow behavior by using the {ref}`new casting operators
+<cvl2-casting>`.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Comparisons require identical types
+
+When comparing two integers using `==`, `<=`, `<`, `>`, or `>=`, CVL 2 will
+require both sides of the equation to have identical types, and {ref}`implicit
+casts <cvl2-casting>` will not be used.  Comparisons with number literals (e.g.
+`0` or `1`) are allowed for any integer type.
+
+If you do not have identical types, the best solution is to use the special
+`to_mathint` operator to convert both sides to `mathint`.  For example:
+
+```cvl
+assert to_mathint(balanceOf(user)) == initial + deposit;
+```
+
+Note that in this example, we do not need to cast the right hand side, since
+the result of `+` is always of type `mathint`.
+
+````{todo}
+When should you not simply cast to `mathint`?  We have one example: consider the
```

Comment:
wait, is this a todo? I say, do! this is good documentation (without the editorializing, obviously)

Reply:
The reason it's left as a TODO is because it's not clear how to generalize from this example.  Are there other situations?  What's the general rule?